src/entities/join/weighted/weightedJoin.ts

```diff
@@ -0,0 +1,149 @@
+import { createPublicClient, encodeFunctionData, http } from 'viem';
+import { Token, TokenAmount, WeightedEncoder } from '../../..';
+import { Address } from '../../../types';
+import {
+    BALANCER_HELPERS,
+    BALANCER_VAULT,
+    CHAINS,
+    ZERO_ADDRESS,
+} from '../../../utils';
+import { balancerHelpersAbi, vaultAbi } from '../../../abi';
+import {
+    checkInputs,
+    getAmountsIn,
+    getAssets,
+    getJoinParameters,
+    getUserData,
+} from './helpers';
+import {
+    BaseJoin,
+    JoinCallInput,
+    JoinInput,
+    JoinKind,
+    JoinQueryResult,
+    PoolState,
+} from '..';
+
+export class WeightedJoin implements BaseJoin {
+    // TODO - Probably not needed
+    getInstance(): WeightedJoin {
+        return new WeightedJoin();
+    }
+
+    public async query(
+        input: JoinInput,
+        poolState: PoolState,
+    ): Promise<JoinQueryResult> {
+        // TODO - This would need extended to work with relayer
+
+        checkInputs(input, poolState);
+
+        const poolAssets = poolState.tokens.map((t) => t.address);
+        const assets = getAssets(input, poolAssets);
```

Comment:
you mean the `getAssets` method? it's there to make sure we use ZERO_ADDRESS when needed
if we do go down the path of having a `joinWithNativeAsset`, this will likely not be needed anymore

Reply:
Oh, I get what you mean. Since it's a query, it doesn't matter if we're doing with native/wrapped address.
I do use the result of that method to build `JoinQueryResult` that `query` returns (which is then relevant to build the transaction itself) but as I said, this will all likely change with the new approach.